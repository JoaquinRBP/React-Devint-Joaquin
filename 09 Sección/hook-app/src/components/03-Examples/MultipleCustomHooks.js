import React from 'react';
import '../02-useEffect/effects.css';
import {useFetch} from '../../hooks/useFetch';
import {useCounter} from '../../hooks/useCounter';
export const MultipleCustomHooks =()=>{
    const {counter, increment} = useCounter(1);
    const {loading, data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
    const {author, quote} = !!data&&data[0];
    return (
        <>
            <h1>Custom Hooks</h1>
            
            {loading
            ? (<div className="alert alert-infor text-center">
                Loading...
            </div>)
            :(
                <blockquote className="blockquote text-right">
                <a className="mb-0">{quote}</a>
                <footer className="blockqueote-footer">
                    {author}
                </footer>
            </blockquote>
            )
            }
            <button 
            className="btn btn-primary"
            onClick={increment}> Siguiente Autor</button>
        </>
    )
}