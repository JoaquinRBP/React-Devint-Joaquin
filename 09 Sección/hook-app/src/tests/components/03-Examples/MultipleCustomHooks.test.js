import React from 'react';
import {shallow} from 'enzyme';
import {MultipleCustomHooks} from '../../../components/03-Examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import {useCounter} from '../../../hooks/useCounter';
jest.mock('../../../hooks/useFetch');
// jest.mock('../../../hooks/useCounter');
describe('Pruebas sobre MultipleCustomHooks',()=>{
    // useCounter.mockReturnValue({
    //     counter: 10,
    //     increment: () => {}
    // });
    test('debe de mostrarse correctamente',()=>{
        useFetch.mockReturnValue({
            data:null,
            loading:false,
            error:null
        })
        const wrapper = shallow(<MultipleCustomHooks />)
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar la informaciÃ³n', () => {
        useFetch.mockReturnValue({
            data:[{
                author:'Joaquin',
                quote: 'Hola Pussy Destroyer'
            }],
            loading:false,
            error:null
        });
        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('Hola Pussy Destroyer');
        expect(wrapper.find('footer').text().trim()).toBe('Joaquin');
    });
    
})