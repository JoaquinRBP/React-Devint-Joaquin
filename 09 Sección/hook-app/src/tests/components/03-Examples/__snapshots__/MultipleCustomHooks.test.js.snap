// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre MultipleCustomHooks debe de mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    Custom Hooks
  </h1>
  <blockquote
    className="blockquote text-right"
  >
    <a
      className="mb-0"
    />
    <footer
      className="blockqueote-footer"
    />
  </blockquote>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
     Siguiente Autor
  </button>
</Fragment>
`;
